src/lib.rs

```diff
@@ -524,23 +549,30 @@ pub fn finish_data(data: &[u8]) -> ! {
     }
 }
 
-pub fn storage_load(key: &[u8; 32]) -> [u8; 32] {
-    let mut ret = [0u8; 32];
+pub fn storage_load(key: &StorageKey) -> StorageValue {
+    let ret = StorageValue::default();
 
     unsafe {
-        native::ethereum_storageLoad(key.as_ptr() as *const u32, ret.as_mut_ptr() as *const u32);
+        native::ethereum_storageLoad(
+            key.bytes.as_ptr() as *const u32,
+            ret.bytes.as_ptr() as *const u32,
```

Comment:
This needs to be `as_mut_ptr`.

Reply:
it is in an unsafe block so it doesn't matter, but will change it.